Framework/Crystal/src/CentroidPeaks.cpp

```diff
@@ -357,5 +336,23 @@ int CentroidPeaks::findPixelID(std::string bankName, int col, int row) {
   }
 }
 
+void CentroidPeaks::removeEdgePeaks(
+    Mantid::DataObjects::PeaksWorkspace *peakWS) {
+  int Edge = getProperty("EdgePixels");
+  std::vector<int> badPeaks;
+  for (int i = int(peakWS->getNumberPeaks()) - 1; i >= 0; --i) {
+    // Get a direct ref to that peak.
+    auto &peak = peakWS->getPeak(i);
```

Comment:
Make this a 
`const auto &peak` so that we know its impossible to accidentally change the value as we take it by ref